core/ir/instr_api.h

```diff
@@ -2337,6 +2337,11 @@ DR_API
 instr_t *
 instr_create_pusha(void *drcontext);
 
+DR_API
+/** Returns whether \p instr loads mcontext base to some reg. */
+bool
+instr_is_load_mcontext_base(instr_t *instr);
```

Comment:
Is it okay to expose this information in a public API? If not, then the alternative is that we pass the ilist to `instr_is_reg_spill_or_restore_ex` and let it figure out which reg contains mcontext base.